test/tests/functional/pbs_job_array.py

```diff
@@ -287,3 +291,318 @@ def test_qdel_expired_subjob(self):
                             "Error message is not expected")
         else:
             raise self.failureException("subjob in X state can be deleted")
+
+    def test_subjob_comments(self):
+        """
+        Test subjob comments for finished and terminated subjobs
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        j = Job(TEST_USER, attrs={
+            ATTR_J: '1-3', 'Resource_List.select': 'ncpus=1'})
+        j.set_sleep_time(5)
+        j_id = self.server.submit(j)
+        subjid_1 = j.create_subjob_id(j_id, 1)
+        subjid_2 = j.create_subjob_id(j_id, 2)
+        self.server.expect(JOB, {'comment': 'Subjob finished'}, subjid_1)
+        self.server.delete(subjid_2, extend='force')
+        self.server.expect(JOB, {'comment': 'Subjob terminated'}, subjid_2)
+        self.kill_and_restart_svr()
+        self.server.expect(JOB, {'comment': 'Subjob finished'}, subjid_1)
```

Comment:
I think its necessary to have max_attempts=1 as an argument to the expect call else there could be a chance of a passing a failed behavior.

Reply:
Are you sure about this @Shrini-h?  Having max_attempts=1 can easily cause a race condition in the test.  It will only check once and if it happens a microsecond after we check, the test will fail.  If you're worried that something will happen in the 30s of default expect() checks, then maybe max_attempts=10?  That'll check for 5s.